File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        //By me FTS and little editorial
        int n=A.size();
        vector<int>ans(n,0);
        vector<int>hasharr(n+1,0);
        int same=0;
        for(int i=0;i<n;i++){
            hasharr[A[i]]++;
            if(hasharr[A[i]]>=2)same++;
            hasharr[B[i]]++;
            if(hasharr[B[i]]>=2)same++;
            ans[i]=same;
        }
        return ans;
    }
};

File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        n = len(A)
        hashA = [0]*(n+1)
        hashB = [0]*(n+1)
        ans = [0]*n
        common = 0
        for i in range(n):
            hashA[A[i]] = 1
            if hashB[A[i]] == 1:
                common += 1
            hashB[B[i]] = 1
            if hashA[B[i]] == 1:
                common += 1
            ans[i] = common
        return ans
            
            